# UK House Price Index_EDA

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
# Loading the Dataset

data = pd.read_csv('C:/Users/HP/Desktop/Krishna/Project 10/UK-HPI-full-file-2017-01.csv')
data.head(5)

Date     RegionName   AreaCode  AveragePrice      Index  IndexSA  \
0  01/01/2004  Aberdeenshire  S12000034   81693.66964  40.864214      NaN   
1  01/02/2004  Aberdeenshire  S12000034   81678.76231  40.856757      NaN   
2  01/03/2004  Aberdeenshire  S12000034   83525.09702  41.780317      NaN   
3  01/04/2004  Aberdeenshire  S12000034   84333.67900  42.184780      NaN   
4  01/05/2004  Aberdeenshire  S12000034   86379.95396  43.208353      NaN   

   1m%Change  12m%Change  AveragePriceSA  SalesVolume  ...  FOO1m%Change  \
0        NaN         NaN             NaN        388.0  ...           NaN   
1  -0.018248         NaN             NaN        326.0  ...           NaN   
2   2.260483         NaN             NaN        453.0  ...           NaN   
3   0.968071         NaN             NaN        571.0  ...           NaN   
4   2.426403         NaN             NaN        502.0  ...           NaN   

   FOO12m%Change     NewPrice   NewIndex  New1m%Change  New12m%Change  \
0            NaN  88436.13953  40.267250           NaN            NaN   
1            NaN  88606.44649  40.344795      0.192576            NaN   
2            NaN  90296.91375  41.114508      1.907838            NaN   
3            NaN  90319.87844  41.124964      0.025432            NaN   
4            NaN  91989.17763  41.885039      1.848208            NaN   

      OldPrice   OldIndex  Old1m%Change  Old12m%Change  
0  81043.95084  40.883367           NaN            NaN  
1  80965.29542  40.843688     -0.097053            NaN  
2  82903.23948  41.821302      2.393549            NaN  
3  84003.99161  42.376586      1.327755            NaN  
4  86222.73484  43.495852      2.641235            NaN  

[5 rows x 50 columns]

In [13]:
print('Columns of the data: ', '\n\n',data.columns)
print('\n')
print('Length of the columns: ', len(data.columns))
print('\n')
print('Shape of the data: ', data.shape)
print('\n')
print('Dataset Information: ',data.info())

Columns of the data:  

 Index(['Date', 'RegionName', 'AreaCode', 'AveragePrice', 'Index', 'IndexSA',
       '1m%Change', '12m%Change', 'AveragePriceSA', 'SalesVolume',
       'DetachedPrice', 'DetachedIndex', 'Detached1m%Change',
       'Detached12m%Change', 'SemiDetachedPrice', 'SemiDetachedIndex',
       'SemiDetached1m%Change', 'SemiDetached12m%Change', 'TerracedPrice',
       'TerracedIndex', 'Terraced1m%Change', 'Terraced12m%Change', 'FlatPrice',
       'FlatIndex', 'Flat1m%Change', 'Flat12m%Change', 'CashPrice',
       'CashIndex', 'Cash1m%Change', 'Cash12m%Change', 'MortgagePrice',
       'MortgageIndex', 'Mortgage1m%Change', 'Mortgage12m%Change', 'FTBPrice',
       'FTBIndex', 'FTB1m%Change', 'FTB12m%Change', 'FOOPrice', 'FOOIndex',
       'FOO1m%Change', 'FOO12m%Change', 'NewPrice', 'NewIndex', 'New1m%Change',
       'New12m%Change', 'OldPrice', 'OldIndex', 'Old1m%Change',
       'Old12m%Change'],
      dtype='object')


Length of the columns:  50


Shape of the data:  (11163

### Data Cleaning

In [14]:
data.isnull().sum()

Date                           0
RegionName                     0
AreaCode                       0
AveragePrice                   0
Index                          0
IndexSA                   107994
1m%Change                    465
12m%Change                  5292
AveragePriceSA            107994
SalesVolume                  882
DetachedPrice               1860
DetachedIndex               1860
Detached1m%Change           2291
Detached12m%Change          7008
SemiDetachedPrice           1860
SemiDetachedIndex           1860
SemiDetached1m%Change       2291
SemiDetached12m%Change      7008
TerracedPrice               1844
TerracedIndex               1844
Terraced1m%Change           2275
Terraced12m%Change          6992
FlatPrice                   1595
FlatIndex                   1595
Flat1m%Change               2027
Flat12m%Change              6755
CashPrice                  85525
CashIndex                  85525
Cash1m%Change              85953
Cash12m%Change             90661
MortgagePr

In [15]:
data_N = data.drop(['IndexSA','AveragePriceSA'], axis=1)
data_N.shape
# Shape of the data after removing two columns

(111633, 48)

In [30]:
Data_C = data_N.dropna(axis=0, how='any',inplace=False)
Data_C.head(5)
# Removing the rows where there is null values

Date     RegionName   AreaCode  AveragePrice      Index  1m%Change  \
108  01/01/2013  Aberdeenshire  S12000034   170639.7192  85.356160  -0.309562   
109  01/02/2013  Aberdeenshire  S12000034   172926.8599  86.500217   1.340333   
110  01/03/2013  Aberdeenshire  S12000034   173254.1927  86.663953   0.189290   
111  01/04/2013  Aberdeenshire  S12000034   174269.6560  87.171901   0.586112   
112  01/05/2013  Aberdeenshire  S12000034   174656.0565  87.365183   0.221726   

     12m%Change  SalesVolume  DetachedPrice  DetachedIndex  ...  FOO1m%Change  \
108   -2.763005        261.0    236804.1417      84.311011  ...     -0.331027   
109    0.444229        241.0    239929.4073      85.423721  ...      1.362826   
110    1.546716        383.0    240684.3117      85.692495  ...      0.247330   
111    1.287708        389.0    242471.1113      86.328661  ...      0.615045   
112   -0.232841        513.0    242134.2998      86.208744  ...      0.102216   

     FOO12m%Change     NewPrice   NewIndex  New1m%Change  New12m%Change  \
108      -2.357107  183969.0424  83.765838     -1.072152      -2.852129   
109       1.179455  185247.9826  84.348173      0.695193       0.061305   
110       1.570631  186403.4054  84.874266      0.623717       1.385895   
111       1.407407  189231.7605  86.162089      1.517330       2.048274   
112      -0.173931  189857.2634  86.446897      0.330549       0.527697   

        OldPrice   OldIndex  Old1m%Change  Old12m%Change  
108  169309.4966  85.409733     -0.266558      -2.754829  
109  171650.3277  86.590587      1.382575       0.471371  
110  171972.1848  86.752951      0.187507       1.585057  
111  172918.5518  87.230354      0.550302       1.290719  
112  173294.6386  87.420075      0.217494      -0.228688  

[5 rows x 48 columns]

In [31]:
Data_C.shape
# Shape of the dataset after removing the null values

(20058, 48)

In [33]:
Data_C.isnull().sum()
# The dataset is clean

Date                      0
RegionName                0
AreaCode                  0
AveragePrice              0
Index                     0
1m%Change                 0
12m%Change                0
SalesVolume               0
DetachedPrice             0
DetachedIndex             0
Detached1m%Change         0
Detached12m%Change        0
SemiDetachedPrice         0
SemiDetachedIndex         0
SemiDetached1m%Change     0
SemiDetached12m%Change    0
TerracedPrice             0
TerracedIndex             0
Terraced1m%Change         0
Terraced12m%Change        0
FlatPrice                 0
FlatIndex                 0
Flat1m%Change             0
Flat12m%Change            0
CashPrice                 0
CashIndex                 0
Cash1m%Change             0
Cash12m%Change            0
MortgagePrice             0
MortgageIndex             0
Mortgage1m%Change         0
Mortgage12m%Change        0
FTBPrice                  0
FTBIndex                  0
FTB1m%Change              0
FTB12m%Change       

### Exploratory Data Analysis - EDA

In [34]:
Data_C.head(5)
# Sample data from the cleaned dataset

Date     RegionName   AreaCode  AveragePrice      Index  1m%Change  \
108  01/01/2013  Aberdeenshire  S12000034   170639.7192  85.356160  -0.309562   
109  01/02/2013  Aberdeenshire  S12000034   172926.8599  86.500217   1.340333   
110  01/03/2013  Aberdeenshire  S12000034   173254.1927  86.663953   0.189290   
111  01/04/2013  Aberdeenshire  S12000034   174269.6560  87.171901   0.586112   
112  01/05/2013  Aberdeenshire  S12000034   174656.0565  87.365183   0.221726   

     12m%Change  SalesVolume  DetachedPrice  DetachedIndex  ...  FOO1m%Change  \
108   -2.763005        261.0    236804.1417      84.311011  ...     -0.331027   
109    0.444229        241.0    239929.4073      85.423721  ...      1.362826   
110    1.546716        383.0    240684.3117      85.692495  ...      0.247330   
111    1.287708        389.0    242471.1113      86.328661  ...      0.615045   
112   -0.232841        513.0    242134.2998      86.208744  ...      0.102216   

     FOO12m%Change     NewPrice   NewIndex  New1m%Change  New12m%Change  \
108      -2.357107  183969.0424  83.765838     -1.072152      -2.852129   
109       1.179455  185247.9826  84.348173      0.695193       0.061305   
110       1.570631  186403.4054  84.874266      0.623717       1.385895   
111       1.407407  189231.7605  86.162089      1.517330       2.048274   
112      -0.173931  189857.2634  86.446897      0.330549       0.527697   

        OldPrice   OldIndex  Old1m%Change  Old12m%Change  
108  169309.4966  85.409733     -0.266558      -2.754829  
109  171650.3277  86.590587      1.382575       0.471371  
110  171972.1848  86.752951      0.187507       1.585057  
111  172918.5518  87.230354      0.550302       1.290719  
112  173294.6386  87.420075      0.217494      -0.228688  

[5 rows x 48 columns]

In [44]:
Data_C['RegionName'].value_counts()

# There are information about 427 unique region 

Nottinghamshire                  47
Medway                           47
Woking                           47
East Devon                       47
Ealing                           47
West Dorset                      47
Hammersmith and Fulham           47
King's Lynn and West Norfolk     47
South Oxfordshire                47
Mid Sussex                       47
West Lancashire                  47
South Somerset                   47
East Hertfordshire               47
Sandwell                         47
Oxford                           47
Midlothian                       47
Inner London                     47
Taunton Deane                    47
Kettering                        47
Castle Point                     47
Nuneaton and Bedworth            47
Scarborough                      47
Oxfordshire                      47
Bolton                           47
Swale                            47
Highland                         47
Barrow-in-Furness                47
Ashfield                    

In [47]:
Data_C.describe()
# Descriptive Statistics

AveragePrice         Index     1m%Change    12m%Change    SalesVolume  \
count  2.005800e+04  20058.000000  20058.000000  20058.000000   20058.000000   
mean   2.150876e+05    100.176733      0.508485      5.741983    1045.368980   
std    1.187670e+05      8.805195      1.431943      4.998800    6564.014131   
min    6.242030e+04     69.927852    -15.474471    -19.578963       1.000000   
25%    1.388902e+05     94.344225     -0.285609      2.403973     133.000000   
50%    1.867402e+05    100.000000      0.521618      5.270829     197.000000   
75%    2.531936e+05    105.407050      1.315809      8.694426     321.000000   
max    1.365050e+06    135.278859     17.891853     35.752892  128272.000000   

       DetachedPrice  DetachedIndex  Detached1m%Change  Detached12m%Change  \
count   2.005800e+04   20058.000000       20058.000000        20058.000000   
mean    3.721039e+05     100.209825           0.540530            5.970802   
std     3.026384e+05       9.036924           1.456233            5.016838   
min     1.041172e+05      68.960612         -15.408105          -19.550909   
25%     2.150106e+05      94.032600          -0.246777            2.650698   
50%     2.868868e+05     100.000000           0.542347            5.520758   
75%     4.086189e+05     105.731300           1.363262            8.951497   
max     3.595369e+06     137.271632          17.942573           34.636264   

       SemiDetachedPrice  ...  FOO1m%Change  FOO12m%Change      NewPrice  \
count       2.005800e+04  ...  20058.000000   20058.000000  2.005800e+04   
mean        2.401891e+05  ...      0.512717       5.769417  2.472889e+05   
std         2.423402e+05  ...      1.435373       4.982899  1.124331e+05   
min         6.945813e+04  ...    -15.437787     -22.917518  6.994290e+04   
25%         1.357741e+05  ...     -0.281851       2.476263  1.719791e+05   
50%         1.826444e+05  ...      0.522439       5.307003  2.207836e+05   
75%         2.599537e+05  ...      1.320115       8.716236  2.867761e+05   
max         3.672499e+06  ...     17.930866      34.381412  1.126122e+06   

           NewIndex  New1m%Change  New12m%Change      OldPrice      OldIndex  \
count  20058.000000  20058.000000   20058.000000  2.005800e+04  20058.000000   
mean     100.513408      0.587954       6.627232  2.127120e+05    100.142306   
std        9.768721      1.763460       5.481739  1.194326e+05      8.735693   
min       69.569230    -15.054120     -18.732816  6.204307e+04     69.941265   
25%       94.300163     -0.402270       2.883913  1.369083e+05     94.339543   
50%      100.000000      0.530284       6.028143  1.840838e+05    100.000000   
75%      105.251508      1.463667       9.929294  2.501041e+05    105.379586   
max      140.720103     17.164233      39.094299  1.384330e+06    135.093923   

       Old1m%Change  Old12m%Change  
count  20058.000000   20058.000000  
mean       0.502526       5.663930  
std        1.434042       4.996249  
min      -15.483818     -19.620960  
25%       -0.296535       2.323979  
50%        0.516097       5.183976  
75%        1.318133       8.621315  
max       17.908175      35.731772  

[8 rows x 45 columns]

- Shows the Descriptive statistics of the dataset
- Mean average house price for a geography in a particular period is 215,087 euros
- Minimum average house price for a geography in a particular period is 62,420 euros
- Mean Number of registered transactions for a geography in a particular period is 1,045, where the minimum goes to even 1 and maximum to 128,272 (This is an outlier in our dataset)
- In case of house price index, mean value is 100, minimum is 69 and maximum is 135

In [84]:
Tab_1 = Data_C.groupby(['RegionName']).Index.agg([np.max,np.min,np.mean])
Tab_1.sort_values('mean', ascending=False).head(10)

amax       amin        mean
RegionName                                          
Richmondshire      115.254315  93.143331  105.945106
Luton              135.278859  86.963121  104.951491
West Lancashire    114.034752  98.082384  104.742542
South Oxfordshire  120.808534  92.113601  104.700519
East Renfrewshire  121.293908  93.922195  104.327012
Daventry           122.068747  91.286511  103.804650
Watford            133.926896  83.992641  103.801269
Malvern Hills      116.149349  93.437479  103.737091
Eden               111.006622  95.957344  103.698220
North Lanarkshire  114.564265  96.199535  103.556686

### Top 10 regions where the house price index for a geography in a particular period is high (in-terms of mean value)

- Richmondshire
- Luton
- West Lancashire
- South Oxfordshire
- East Renfrewshire
- Daventry
- Watford	
- Malvern Hills
- Eden
- North Lanarkshire

In [85]:
Tab_2 = Data_C.groupby(['RegionName']).SalesVolume.agg([np.max,np.min,np.mean])
Tab_2.sort_values('mean', ascending=False).head(10)

amax     amin          mean
RegionName                                           
Great Britain         128272.0  49115.0  82238.851064
England and Wales     117255.0  44463.0  74429.063830
England               111901.0  42605.0  70921.382979
South East             20152.0   7850.0  13224.085106
London                 16322.0   4616.0   9433.851064
East of England        13804.0   5416.0   8962.808511
North West             13516.0   4729.0   8526.914894
South West             13379.0   4764.0   8397.936170
Scotland               11017.0   4285.0   7784.404255
West Midlands Region   10566.0   3608.0   6575.170213

### Top 10 regions where the number of registered transactions for a geography in a particular period is high (in-terms of mean value)

- Great Britain
- England and Wales
- England
- South East
- London
- East of England
- North West
- South West
- Scotland
- West Midlands Region

In [119]:
Tab_3 = Data_C.groupby(['RegionName'])['DetachedPrice','SemiDetachedPrice','TerracedPrice','FlatPrice'].agg([np.mean])
Tab_3.head(10)

DetachedPrice SemiDetachedPrice  TerracedPrice      FlatPrice
                          mean              mean           mean           mean
RegionName                                                                    
Aberdeenshire    268254.482947     166157.064349  139026.115026  108912.445413
Adur             381913.754515     280530.153906  242579.406960  161570.628657
Allerdale        203771.544079     135185.870294  114399.438298   93541.330565
Amber Valley     205485.564379     129096.913509  108818.949611   90701.075479
Angus            217186.152502     141195.364515  103283.111805   81941.788709
Argyll and Bute  196644.523362     126741.929251   99482.797184   85804.874164
Arun             372262.580540     240552.733743  199530.385964  141525.843679
Ashfield         160884.872511     105747.172642   87955.046011   66342.259967
Ashford          362945.576881     222305.441923  186557.870889  135910.363264
Aylesbury Vale   443657.654509     266804.744619  221376.610185  154232.807385

- Mean price of different property type such as Detached, Semi-Detached, Terraced and Flat with respect to the region is given in the above table
- It can be sorted in terms property type, which is of focus and find which region is having the highest and lowest prices 